tests/integration/json_test.go

```diff
@@ -1,114 +0,0 @@
-package integration
```

Comment:
why was this removed?

Reply:
@kadel @cdrage ```-o json``` is tested as a flag to supported command but should not be a test file. For example 

we have already a test file ```cmd_app_test``` where most of the app related command is validated. So it's better to test the ```-o json``` flag along with this test file instead of making the same scenario in a different file along with the ```-o json``` flag. See - https://github.com/openshift/odo/blob/master/tests/integration/cmd_app_test.go#L69 where it has been implemented.

Advantage - 
1. No duplicate scenario in separate file
2. Reduces total test time
3. Later it will be easy to manage and maintain json flag test along with the supported command

So i removed the file but not its scenario. I add all the ```-o json``` flag scenario in the respective test file.

Reply:
IMO it's nice to have it in it's own machine-readable-output testing file... it's an essential part of `odo` and having to go through a bunch of other files just to find where to put `-o json` tests would be cumbersome. I agree with @kadel  that we should not be deleting this file.

Reply:
> IMO it's nice to have it in it's own machine-readable-output testing file... it's an essential part of odo and having to go through a bunch of other files just to find where to put -o json tests would be cumbersome. I agree with @kadel that we should not be deleting this file.

Right now keeping the test in separate file looks good because only few command/subcommands support machine readable output. The size of the file and duplicate scenarios will be increased for sure once all the task of issue https://github.com/openshift/odo/issues/1739 is fixed

The way to handle it smartly without duplicating the scenario in separate file

cmd_app_test.go
```
Context("when running app command without app parameter in directory that contains .odo config directory", func() {
	It("should successfully execute list, describe and delete", func() {
	    helper.CopyExample(filepath.Join("source", "nodejs"), context)
		helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--app", appName, "--project", project, "--context", context)
		helper.CmdShouldPass("odo", "push", "--context", context)

		appListOutput := helper.CmdShouldPass("odo", "app", "list")
		Expect(appListOutput).To(ContainSubstring(appName))
		
		helper.CmdShouldPass("odo", "app", "describe")
		
		helper.CmdShouldPass("odo", "app", "delete", "-f")
	})
})
```
Let see how the machine readable output will be add without hurting the core app test
```
Context("when running app command without app parameter in directory that contains .odo config directory", func() {
	It("should successfully execute list, describe and delete along with machine readable output", func() {
	    helper.CopyExample(filepath.Join("source", "nodejs"), context)
		helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--app", appName, "--project", project, "--context", context)
		helper.CmdShouldPass("odo", "push", "--context", context)

		appListOutput := helper.CmdShouldPass("odo", "app", "list")
		Expect(appListOutput).To(ContainSubstring(appName))
		actualCompListJSON := helper.CmdShouldPass("odo", "list", "-o", "json")
		desiredCompListJSON := fmt.Sprintf(`{"kind":"List","apiVersion":"odo.openshift.io/v1alpha1","metadata":{},"items":[{"kind":"Component","apiVersion":"odo.openshift.io/v1alpha1","metadata":{"name":"nodejs","creationTimestamp":null, "namespace":"%s"},"spec":{"type":"nodejs","app":"app","source":"file://./"},"status":{"state":"Pushed"}}]}`, project)
		Expect(desiredCompListJSON).Should(MatchJSON(actualCompListJSON))

		helper.CmdShouldPass("odo", "app", "describe")
		desiredDesAppJSON := fmt.Sprintf(`{"kind":"app","apiVersion":"odo.openshift.io/v1alpha1","metadata":{"name":"myapp","namespace":"%s","creationTimestamp":null},"spec":{}}`, project)
		actualDesAppJSON := helper.CmdShouldPass("odo", "app", "describe", "myapp", "-o", "json")
		Expect(desiredDesAppJSON).Should(MatchJSON(actualDesAppJSON))

		helper.CmdShouldPass("odo", "app", "delete", "-f")
	})
})
```

Moreover it will be easy to test ```-o``` flag on those existing variety of scenarios of supported command.

Reply:
@amitkrout got a point there. It might be good to have the json output in a test together with the command. 
 - It will reduce the number of tests
 - It makes logical sense, you have only one test setup, and than just execute both commands (with `-o` and  without `-o`)
 - Json output doesn't need any special place, it is just a regular command with different output format.